xonsh/ptk_shell/shell.py

```diff
@@ -64,8 +67,14 @@ def __init__(self, **kwargs):
         self.history = ThreadedHistory(PromptToolkitHistory())
         self.prompter = PromptSession(history=self.history)
         self.pt_completer = PromptToolkitCompleter(self.completer, self.ctx, self)
-        self.key_bindings = KeyBindings()
-        load_xonsh_bindings(self.key_bindings)
+        self.key_bindings = merge_key_bindings(
+            [
+                load_xonsh_bindings(),
+                # Unless we are in VI_MODE, we are in EMACS_MODE
+                load_emacs_shift_selection_bindings(),
```

Comment:
I don't understand the above comment. The code does not really differentiate between Vi/Emacs modes.

Reply:
ha fair enough! it was a reference to `load_emacs_shift_selection_bindings` only being available in EMACS_MODE, but that's probably pretty intuitive by the name